### PR TITLE
[Docs] Update configuring_logging_links_in_the_ui.md

### LINKED ISSUE #5760: [Docs] Update logging links documentation for Cloudwatch
### Description

The current [docs page](https://docs.flyte.org/en/latest/user_guide/productionizing/configuring_logging_links_in_the_ui.html#configure-logging) includes an example that is generic and uses outdated configuration:

```
task_logs:
  plugins:
    logs:
      templates:
        - displayName: <name-to-show>
          templateUris:
            - "https://console.aws.amazon.com/cloudwatch/home?region=us-east-1#logEventViewer:group=/flyte-production/kubernetes;stream=var.log.containers.{{.podName}}_{{.namespace}}_{{.containerName}}-{{.containerId}}.log"
            - "https://some-other-source/home?region=us-east-1#logEventViewer:group=/flyte-production/kubernetes;stream=var.log.containers.{{.podName}}_{{.namespace}}_{{.containerName}}-{{.containerId}}.log"
          messageFormat: 0 # this parameter is optional, but use 0 for "unknown", 1 for "csv", or 2 for "json
```

Considering the notes and fields on the current struct:
https://github.com/flyteorg/flyte/blob/0eaa20e8b92245555be8fd725f97e52fbed54e78/flyteplugins/go/tasks/logs/config.go#L11-L34

The Cloudwatch example should use the supported fields and a baseline`templateUri` structure, indicating the prerequisites for log links to work on EKS ( [Cloudwatch Observability for EKS add-on](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/Container-Insights-setup-EKS-addon.html)).

Also the base Helm `values` should include the config structure, with logging disabled by default.

---

### Are you sure this issue hasn't been raised already?

- [X] Yes

### Have you read the Code of Conduct?

- [X] Yes

### PR DESCRIPTION
**Pull request by pranshustuff**
Added log templates for AWS CloudWatch, GCP Stackdriver, and Kubernetes logs with their templateUris in new example.

Also included prerequisites for CloudWatch Observability on EKS.

## Tracking issue
Closes #5760 

## Why are the changes needed?
The current example is generic and uses outdated configuration:

## What changes were proposed in this pull request?

I am just changing the example to be more flexible, up to date and with more specific templateUris.

## Check all the applicable boxes 

- [x] I updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] All commits are signed-off.

## Docs link
https://docs.flyte.org/en/latest/user_guide/productionizing/configuring_logging_links_in_the_ui.html#configure-logging

### CODE REVIEW

File: docs/user_guide/productionizing/configuring_logging_links_in_the_ui.md [hunk 1]
```diff
@@ -58,11 +58,20 @@ task_logs:
   plugins:
     logs:
       templates:
-        - displayName: <name-to-show>
+        - displayName: AWS CloudWatch Logs
+          templateUris:
```
Comment (davidmirror-ops) [inline docs/user_guide/productionizing/configuring_logging_links_in_the_ui.md:62]:
```suggestion
          cloudwatch-enabled: true
          cloudwatch-template-uri: "https://console.aws.amazon.com/cloudwatch/home?region=<MY_AWS_REGION>#logsV2:log-groups/log-group/$252Faws$252Fcontainerinsights$252F<MY_EKS_CLUSTER_NAME>$252Fapplication$3FlogStreamNameFilter$3Dvar.log.containers.{{`{{.podName}}`}}_{{`{{.namespace}}`}}_{{`{{.containerName}}`}}" 
```

File: docs/user_guide/productionizing/configuring_logging_links_in_the_ui.md [hunk 2]
```diff
@@ -58,11 +58,20 @@ task_logs:
   plugins:
     logs:
       templates:
```
Comment (davidmirror-ops) [inline docs/user_guide/productionizing/configuring_logging_links_in_the_ui.md:60]:
```suggestion
```
So the YAML structure would be:
```
  plugins:
    logs:
      cloudwatch-enabled: true
      cloudwatch-template-uri:

File: docs/user_guide/productionizing/configuring_logging_links_in_the_ui.md [hunk 3]
```diff
@@ -58,11 +58,20 @@ task_logs:
...
+          prerequisites:
+            - EKS Cluster enabled with CloudWatch Observability Add-on
+            - Ensure the pod emits logs to CloudWatch log streams
+        - displayName: GCP Stackdriver Logs
           templateUris:
-            - "https://console.aws.amazon.com/cloudwatch/home?region=us-east-1#logEventViewer:group=/flyte-production/kubernetes;stream=var.log.containers.{{.podName}}_{{.namespace}}_{{.containerName}}-{{.containerId}}.log"
-            - "https://some-other-source/home?region=us-east-1#logEventViewer:group=/flyte-production/kubernetes;stream=var.log.containers.{{.podName}}_{{.namespace}}_{{.containerName}}-{{.containerId}}.log"
-          messageFormat: 0 # this parameter is optional, but use 0 for "unknown", 1 for "csv", or 2 for "json"
+            - "https://console.cloud.google.com/logs/viewer?project={{.gcpProject}}&resource=k8s_container&advancedFilter=resource.labels.pod_name={{.podName}}&resource.labels.container_name={{.containerName}}&resource.labels.namespace_id={{.namespace}}"
```
Comment (davidmirror-ops) [inline docs/user_guide/productionizing/configuring_logging_links_in_the_ui.md:69]: I haven't tested this one but comparing with a working environment the structure is different. Let's leave this for a different PR

File: docs/user_guide/productionizing/configuring_logging_links_in_the_ui.md [hunk 4]
```diff
@@ -58,11 +58,20 @@ task_logs:
...
+        - displayName: Kubernetes Dashboard Logs
+          templateUris:
+            - "{{.kubernetesUrl}}/namespace/{{.namespace}}/pods/{{.podName}}/logs/{{.containerName}}"
```
Comment (davidmirror-ops) [inline docs/user_guide/productionizing/configuring_logging_links_in_the_ui.md:72]: Again, haven't tried this and couldn't speak about the requirements so let's leave it out from this PR

**Comment by davidmirror-ops** [comment]
@pranshustuff we'd still like to see this contribution land, and would still count towards hacktoberfest. Could you take a look at the comments I left?

**Comment by pranshustuff** [comment]
Yes please, if you can count it towards Hacktober fest that would be great, I'll figure out why it's not passing the unit test and send a few commits by tomorrow.

**Comment by eapolinario** [comment]
This is using the old pattern. We should switch log links to use `Templates` everywhere.
